line_detection/src/brightest-pixel.py

```diff
@@ -0,0 +1,317 @@
+        # self.camera_info_sub = rospy.Subscriber("/camera/camera_info",
+        #                                          CameraInfo,
+        #                                          self.camera_info_callback,
+        #                                          queue_size=1 )
+    
+    # this is what gets called when an image is recieved
+    def image_callback(self, image):
+
+        # use this to record start time for each frame
+        #start_time = time.time()
+        
+        if(self.use_mono and not self.use_compressed_format and image.encoding != 'mono8'):
+            rospy.logerr("image is not mono8! Aborting!")
+            return
+        
+        if self.use_compressed_format:
+            #### direct conversion from ROS CompressedImage to CV2 ####
+            np_arr = np.fromstring(image.data, np.uint8)
+            if self.use_mono:
+                img = cv2.imdecode(np_arr, cv2.CV_LOAD_IMAGE_GRAYSCALE)
+            else:
+                img = cv2.imdecode(np_arr, cv2.CV_LOAD_IMAGE_COLOR)
+        else:
+            #### direct conversion from ROS Image to CV2 ####
+            # first, we need to convert image from sensor_msgs/Image to numpy (or
+            # cv2). For this, we use cv_bridge
+            try:
+                # img = self.bridge.imgmsg_to_cv2(image, "bgr8")
+                img = self.bridge.imgmsg_to_cv2(image,
+                                                desired_encoding="passthrough")
+            except CvBridgeError, e:
+                rospy.logerr(e)
+
+        if img is None:
+            rospy.logerr("error! img is empty!")
+            return
+
+        self.image_height = img.shape[0]
+        self.image_width = img.shape[1]
+
+        # if mono, don't take 3rd dimension since there's only one channel
+        if self.use_mono:
+            roi = img[
+            self.roi_top_left_y:self.roi_top_left_y + self.roi_height,
+            self.roi_top_left_x:self.roi_top_left_x + self.roi_width,
+            ]
+        else:
+            roi = img[
+                self.roi_top_left_y:self.roi_top_left_y + self.roi_height,
+                self.roi_top_left_x:self.roi_top_left_x + self.roi_width,
+                :
+            ]
+
+        # in case roi settings aren't correct, just use the entire image
+        if roi.size <= 0:
+            rospy.logerr("Incorrect roi settings! Will use the entire image instead!")
+            roi = img
+
+        # use entire image as roi (don't cut any parts out)
+        # roi = img
+
+        # get indices of max pixels along each row
+        brightest_pixel_indices = np.argmax(roi, axis=1)
+        # get the values of max pixels along each row
+        brightest_pixel_values = np.amax(roi, axis=1)
+
+        # make an empty image
+        brightest_pixels = np.zeros(roi.shape)
+
+        count = 0
+        # now fill the image only with the brightest_pixels
+        for index in brightest_pixel_indices:
```

Comment:
for row, (col, pix) in enumerate(zip(brightest_pixel_indices, brightest_pixel_values):
        brightest_pixels[row, col] = pix

Never use a count mechanism when using a for loop; use enumerate.
Use better variable names; especially in loops.
Use the item; versus looking up the item. 
The code will run faster and is easier to figure out!

[Transforming Code into Beautiful, Idiomatic Python] (https://www.youtube.com/watch?v=OSGv2VnC0go&index=3&list=PL7xwbQW5nwgw_E5OxYy01xJxzZIUpQxBN)